core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -662,7 +662,7 @@ sealed trait ZIO[-R, +E, +A]
     ev1: A IsSubtypeOfOutput ZIO[R1, E1, B],
     trace: Trace
   ): ZIO[R1, E1, B] =
-    self.flatMap(a => ev1(a))
+    self.asInstanceOf[UIO[ZIO[R1, E1, B]]].flatMap(ZIO.identityFn)
```

Comment:
```suggestion
    self.flatMap(ev1)
```
should be enough to avoid the extra allocation?

Reply:
Nice one, you're right 🙏

Reply:
Very weird how `self.flatMap(ev1)` and `self.flatMap(v => ev1(v))` end up with different byte code 😕